go/logger/rpc_adapters_rpc.go

```diff
@@ -22,28 +22,131 @@ func (l LogOutputWithDepthAdder) CloneWithAddedDepth(
 	return LogOutputWithDepthAdder{l.Logger.CloneWithAddedDepth(depth)}
 }
 
-// LogOutputWithDebugAndInfoGuards is a wrapper around Logger that disables
-// Info and Debug logging.
-type LogOutputWithDebugAndInfoGuards struct {
-	Logger
+// logOutputWithInfoToDebugShifter is a wrapper around l that shifts all
+// Info logging to Debug. This is used in client/service to prevent
+// rpc/connection.go logs from leaking to CLI.
+type logOutputWithInfoToDebugShifter struct {
+	l Logger
```

Comment:
If you embed this `Logger` then you don't need to implement all the non-shifted methods.

Reply:
We’d still need to implement all logging related ones because of the added depth in the constructor, and adding depth is required for overriding Info and CInfof. Since that’s most of methods, I figured I’d just make it a field so that we have added benefit of not forgetting any of the logging methods.